header.go

```diff
@@ -1508,9 +1513,7 @@ func (h *ResponseHeader) AppendBytes(dst []byte) []byte {
 
 	for i, n := 0, len(h.h); i < n; i++ {
 		kv := &h.h[i]
-		if !bytes.Equal(kv.key, strDate) {
-			dst = appendHeaderLine(dst, kv.key, kv.value)
-		}
+		dst = appendHeaderLine(dst, kv.key, kv.value)
```

Comment:
I don't think these lines should be changed. Before setting your own `Date` header would always get ignored. With this change it will be included which can cause duplicate `Date` headers.
You could do `if noDefaultDate || !bytes.Equal(kv.key, strDate) {` if you think people should be able to set their own `Date` header if they set `NoDefaultDate` to `true`.

Reply:
The `Date` header will not get duplicated because:
* if `server.NoDefaultDate` is set, then the server won't add the `Date` header itself, as per:
https://github.com/valyala/fasthttp/pull/629/files#diff-5e68413c67a51bcc7bf6e42ec37bde13L1496-R1501
* if one insists on adding a `Date` header, one can use the `headers.Add("Date", "...")` method, which bypasses the `SetCanonical` method;

The patch I've submitted allows both behaviours:
* either not sending at all the `Date` header, by just setting `server.NoDefaultDate = true`;
* either sending a custom `Date` header by setting again `server.NoDefaultDate = true`, but also calling `response.Headers.Add("Date", "...")`;

After looking through `header.go`, the only way to set the `Date` header is either through `SetCanonical` (which ignores it) or through `Add` (which will allow it to be set);  (and through their alternate forms that in the end call these two variants).

Thus the question is how often one uses the `Add` method;  moreover how often is it used to set the `Date` header.

Reply:
I'm afraid that's not true. A very common case for fashttp is to use it as a proxy, see: https://gist.github.com/erikdubbelboer/a7c55d8a6dd467a3840e35352ccf180c

With your patch this now outputs:
```
HTTP/1.1 200 OK
Server: fasthttp
Content-Length: 0
Date: Mon, 12 Aug 2019 19:51:35 GMT
Date: Mon, 12 Aug 2019 19:51:35 GMT
```

Reply:
OK, then if I use as suggested `if noDefaultDate || !bytes.Equal(kv.key, strDate) { appendHeader(...) }`, then by default the "current" behavior is retained, while by setting `NoDefaultDate = true` will disable both setting the `Date` header by default, but also will not filter the `Date` header anymore.

Would this be OK?

Reply:
That sounds good yes.

Reply:
I've amended my last commit in-line with your suggestion.